system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -54,9 +51,7 @@ bool PeriodicMeasurementNode::setupStart()
     RCLCPP_WARN(this->get_logger(), "setupStart: measurement_timer_ already exists!");
   }
 
-  if (!publish_timer_ &&
-    publish_period_ != PeriodicMeasurementNode::DEFAULT_PUBLISH_WINDOW)
-  {
+  if (!publish_timer_ && publish_period_ > std::chrono::milliseconds(0)) {
```

Comment:
Checking in the start method is too late: this should be checked upon construction.

Reply:
The unit tests are currently deliberately passing in invalid `publish_period`s to avoid this `Collector::clearCurrentMeasurements()` call.

Reply:
Then why not make zero valid and document as such? We only have an issue if the period is strictly less than zero.